win/rl/CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+# TODO(NN): we should rename this to winnle
+add_library(winrl winrl.cc)
+set_target_properties(winrl PROPERTIES CXX_STANDARD 14)
+
+add_custom_target(
+  message_generated COMMAND flatc -o ${NLE_INC_GEN} --cpp
+                            ${nle_SOURCE_DIR}/win/rl/message.fbs)
+
+target_include_directories(
+  winrl
+  PRIVATE ${NLE_INC_GEN} # message_generated.h
+          ${NLE_INC} # hack.h
+          /usr/local/include) # flatbuffers.h
+
+find_library(FLATBUFFERS flatbuffers)
+target_link_libraries(winrl PRIVATE ${FLATBUFFERS})
+add_dependencies(winrl message_generated nledat)
+
+# NOTE: Below here should all be optional (but required for the python
+# interface)
+
+add_subdirectory(${nle_SOURCE_DIR}/third_party/pybind11 pybind11)
+# find_package(pybind11)
+
+# pybind11_add_module(helper EXCLUDE_FROM_ALL helper.cc ${NLE_SRC}/monst.c
+# ${NLE_SRC}/decl.c ${NLE_SRC}/drawing.c)
+pybind11_add_module(helper EXCLUDE_FROM_ALL helper.cc)
+set_target_properties(helper PROPERTIES CXX_STANDARD 14)
+
+target_include_directories(helper PRIVATE ${NLE_INC} ${NLE_INC_GEN})
+# This is a little overkill. We technically only need monst, decl, and drawing
+# target_link_libraries(helper PRIVATE $<TARGET_OBJECTS:nlehack>)
+
+add_library(monst OBJECT ${NLE_SRC}/monst.c)
+target_include_directories(monst PRIVATE ${NLE_INC} ${NLE_INC_GEN})
+add_library(decl OBJECT ${NLE_SRC}/decl.c)
+target_include_directories(decl PRIVATE ${NLE_INC} ${NLE_INC_GEN})
+add_library(drawing OBJECT ${NLE_SRC}/drawing.c)
+target_include_directories(drawing PRIVATE ${NLE_INC} ${NLE_INC_GEN})
+
+target_link_libraries(helper PUBLIC monst decl drawing)
```

Comment:
@heiner I caved in and added pybind11 as a dependency.

I'm a little stuck on trying to link the external objects to helper. Can you give it a look? I've tried various things, but all seem to leave undefined symbols in the `helper.so`.

Reply:
You'll need to link statically.

Reply:
Indeed, but I tried that for a while -- groan, turns out adding `POSITION_INDEPENDENT_CODE ON` as a property to the libs does it.